doc/guides/west/build-flash-debug.rst

```diff
@@ -574,6 +574,11 @@ from :file:`runners/__init__.py`.
    changes break existing test cases, CI testing on upstream pull
    requests will fail.
 
+..
+   Including this automodule fails with a clear warning if west is
+   missing because west_commands/runners/ have an unsurprising
+   dependency on west through (at least) zephyr_ext_common
```

Comment:
Hmm, that's not "right" in the sense that the runners should be a standalone package that only depends on stdlib and third party modules. I see now that you're right in that bossac depends on zephyr_ext_common; that should probably be refactored to move the dependency into the runners package if needed.

But this isn't a dependency on west, though.

Reply:
So are you saying this automodule inclusion could work even when `import west` does not? Sorry this is all completely new to me (and will probably be soon again :-)

Reply:
I'm saying that `runners` does not depend on `west`.

Reply:
> I  see now that you're right in that bossac depends on zephyr_ext_common; that should probably be refactored to move the dependency into the runners package if needed.

So you're saying this: `runners -> bossac -> zephyr_ext_common -> west`
should be replaced by that: `runners ->  zephyr_ext_common -> west` ?



For now:
```python
>>> sys.path.insert(0,os.getenv("ZEPHYR_BASE") + "/scripts/west_commands") # doc/conf.py
>>> import runners
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "ZEPHYR_BASE/scripts/west_commands/runners/__init__.py", line 14, in <module>
    from runners import bossac
  File "ZEPHYR_BASE/scripts/west_commands/runners/bossac.py", line 15, in <module>
    from zephyr_ext_common import ZEPHYR_SCRIPTS
  File "ZEPHYR_BASE/scripts/west_commands/zephyr_ext_common.py", line 15, in <module>
    from west import log
ModuleNotFoundError: No module named 'west'
```

Reply:
> So you're saying this: `runners -> bossac -> zephyr_ext_common -> west`

Ah, OK, I see the issue. That is a bug in the bossac runner then. We should fix it there so it doesn't depend on anything except stdlib or third party; I want to keep `runners` reasonably standalone.

Reply:
> > exposing unexpected dependencies = possible design mistakes

> That is a bug in the bossac runner then

Thanks for the example :-)